backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/SchedulingManager.java

```diff
@@ -578,29 +606,42 @@ private void addPendingResources(VM vm, VDS host, Map<Integer, NumaNodeMemoryCon
 
         Map<Guid, Map<Integer, NumaNodeMemoryConsumption>> result = new HashMap<>();
         for (VM vm : filteredVms) {
-            if (vm.getvNumaNodeList().isEmpty()) {
-                continue;
-            }
+            vmNumaRequirements(vm, nodeAssignment, result);
```

Comment:
ok, I see that you didn't want to introduce not-absolutely-required changes here, but this really complicates the code - I'm ok with changing this code